### ISSUE TITLE
bulkio: Move super-user checks to privileged IO paths

### ISSUE THREAD

**Issue by dt** [cockroachdb/cockroach#52004]
External storage access has historically forced us to require super-user roles to run bulk-jobs. This is because these jobs cause the _server_ to contact external resources and could thus allow a user who does not have access to something to gain access to it _via_ the server (e.g. an internal file server or an S3 bucket that the node's machine account has access to). To mitigate this, we've only allowed super-users who we already trust to run these jobs that can access these APIs. Even then, since even users are network access, we've provided CLI flags that an operator can set to disable this functionality (`--external-io-disable-http` / `--external-io-disable-implicit-auth`).

However this still creates friction for non-root users who just want to import a table -- they can create, they can insert, but they can't IMPORT. With the introduction of `userfile` we're focusing on giving users a secure, internal, permission'ed place to put data that bulk jobs will interact with. We can thus move the super-user only checks from the jobs themselves to the resources on which they act. Of course the jobs will need to check the relevant permissions on the tables(s) they act on (CREATE, INSERT, SELECT, etc). Additionally, we can move the super-user checks into the external storage APIs when and only when those APIs are providing access to something simply by virtue of the server making the request, such as:
 - [x] HTTP + Custom Endpoint overrides (since request is made by server, in server's network, behind firewall, etc)
 - [x] Implicit auth / using the node's machine account
 - [x] Nodelocal (since it is the node's shared filesystem)

Follow-up work could additionally make these individually or collectively a grantable privilege to non-superusers.

Once this is done, and the audit of object permission checks is complete in the job creation paths as well as in the job control paths (i.e. check ownership of individual jobs), we can remove super-user checks from job control and job creation.

**Comment by adityamaru** [comment]
Marking this as closed as all the callouts for IMPORT were implemented. Some future work on having even more fine-grained control over what we allow admin/non-admin users could be picked up, but I don't think it is a priority right now. SQL has already implemented `CONTROLJOB` privilege for granting users the ability to control jobs.
